app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/SpecifyShopType.kt

```diff
@@ -43,10 +42,7 @@ class SpecifyShopType : OsmFilterQuestType<ShopTypeAnswer>() {
 
 
     override fun applyAnswerTo(answer: ShopTypeAnswer, changes: StringMapChangesBuilder) {
-        val otherCheckDateKeys = LAST_CHECK_DATE_KEYS.filterNot { it == SURVEY_MARK_KEY }
-        for (otherCheckDateKey in otherCheckDateKeys) {
-            changes.deleteIfExists(otherCheckDateKey)
-        }
+        changes.deleteOtherCheckDates()
         when (answer) {
             is IsShopVacant -> {
                 changes.deleteIfExists("shop")
```

Comment:
Should this not also delete the survey tag, and therefore the whole quest could just remove all survey tags?
```suggestion
                changes.deleteIfExists("shop")
                changes.deleteIfExists(SURVEY_MARK_KEY)
```

Reply:
> Hm, @matkoniecz wrote this quest.

Okay I'll wait for him to respond too.

> But looking at the code now, it makes no sense to first delete all survey keys except `SURVEY_MARK_KEY` and then delete the `SURVEY_MARK_KEY` anyway.

What about CheckShopType; if we do:
```kotlin
changes.deleteIfExists(SURVEY_MARK_KEY)
changes.addOrModify(SURVEY_MARK_KEY, LocalDate.now().toCheckDateString())
```
Does it compress that down to one change, or should we try and avoid writing code like that?

> So in this case, I guess, writing
> at the very start would be correct because the check date(s) are deleted no matter what the user answers.

I'd probably write a deleteCheckDates function to match the others and use that directly.

Reply:
Well current behavior is to not set the check date if anything else is tagged

Reply:
Ok I have looked at it again, please just write 
```
        for (otherCheckDateKey in LAST_CHECK_DATE_KEYS) {
            changes.deleteIfExists(otherCheckDateKey)
        }

```
and remove the other check date related stuff.

Reply:
Do you mean for SpecifyShopType or CheckShopType.

I'd still rather write a deleteCheckDates function to drop them all, unless you're strongly in favour of the for loop?

Reply:
> Do you mean for SpecifyShopType or CheckShopType.

SpecifyShopType 

> I'd still rather write a deleteCheckDates function to drop them all, unless you're strongly in favour of the for loop?

A helper function is fine. I was referring to the fact that independent of the answer provided in that quest, all check dates should be removed.

Reply:
Done.